datastore/keypairstatusmanager.go

```diff
@@ -0,0 +1,191 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datastore
+
+import (
+	"database/sql"
+	"log"
+)
+
+const createKeypairStatusTableSQL = `
+CREATE TABLE IF NOT EXISTS keypairstatus (
+	id            serial primary key not null,
+	authority_id  varchar(200) not null,
+	key_name      varchar(200) not null,
+	keypair_id    int references keypair null,
+	status        varchar(20)
+)
+`
+
+const createKeypairStatusSQL = "insert into keypairstatus (authority_id,key_name,status) values ($1,$2,$3) RETURNING id"
```

Comment:
nitpicking: why reserved words are in upper and lower case in same sql sentence?

Reply:
...you probably don't want to go there! We've been inconsistent with capitals and lowercase in our SQL. Maybe we should stick with uppercase for reserved words as I think that's the standard, but it doesn't make sense to do that in this PR for all SQL.